doc/examples/filters/plot_j_invariant.py

```diff
@@ -0,0 +1,292 @@
+# Conversion to J-invariance
+# =========================================
+# The function `invariant_denoise` acts as a J-invariant version of a
+# given denoiser. It works by masking a fraction of the pixels, interpolating
+# them, running the original denoiser, and extracting the values returned in
+# the masked pixels. Iterating over the image results in a fully J-invariant
+# output.
+#
+# For any given set of parameters, the J-invariant version of a denoiser
+# is different from the original denoiser, but it is not necessarily better
+# or worse. In the plot below, we see that, for the test image of a cat,
+# the J-invariant version of a wavelet denoiser is significantly better
+# than the original at small values of variance-reduction `sigma` and
+# imperceptibly worse at larger values.
+#
+
+parameters_tested = [{'sigma': sigma, 'convert2ycbcr': True,
+                      'wavelet': 'db2', 'multichannel': True}
+                     for sigma in sigma_range]
+
+denoised_original = [denoise_wavelet(noisy, **params)
+                     for params in parameters_tested]
+
+ground_truth_loss_invariant = [mse(img, image) for img in denoised_invariant]
+ground_truth_loss_original = [mse(img, image) for img in denoised_original]
+
+fig, ax = plt.subplots(figsize=(10, 4))
+
+ax.plot(sigma_range, ground_truth_loss_invariant, color='C0', linestyle='--',
+        label='J-invariant')
+ax.plot(sigma_range, ground_truth_loss_original, color='C1', linestyle='--',
+        label='Original')
+ax.scatter(sigma_range[opt_idx], ground_truth_loss[opt_idx] + 0.001,
+           marker='v', color='red')
+ax.legend()
+ax.set_title(
+    'J-Invariant Denoiser Has Comparable Or '
+    'Better Performance At Same Parameters'
+)
+ax.set_ylabel('MSE')
+ax.set_xlabel('sigma')
+
+#####################################################################
+# Comparing Different Classes of Denoiser
+# =========================================
+# The self-supervised loss can be used to compare different classes of
+# denoiser in addition to choosing parameters for a single class.
+# This allows the user to, in an unbiased way, choose the best parameters
+# for the best class of denoiser for a given image.
+#
+# Below, we show this for an image of the hubble deep field with significant
+# speckle noise added. In this case, the J-invariant calibrated denoiser is
+# better than the default denoiser in each of three families of denoisers --
+# Non-local means, wavelet, and TV norm. Additionally, the self-supervised
+# loss shows that the TV norm denoiser is the best for this noisy image.
+#
+
+image = rgb2gray(img_as_float(hubble_deep_field()[100:250, 50:300]))
+
+sigma = 0.4
+noisy = random_noise(image, mode='speckle', var=sigma ** 2)
+
+parameter_ranges_tv = {'weight': np.arange(0.01, 0.3, 0.02)}
+parameters_tested_tv, losses_tv = calibrate_denoiser_search(
+                                    noisy,
+                                    denoise_tv_chambolle,
```

Comment:
the example raises an error here `TypeError: calibrate_denoiser_search() takes 2 positional arguments but 3 were given`.